src/array/from_geoarrow.rs

```diff
@@ -1,85 +1,94 @@
                     .to_celllistarray(options)
             }
         }
 
-        impl ToCellIndexArray for $array_type {
+        impl<$offset: OffsetSizeTrait> ToCellIndexArray for $array_type {
             fn to_cellindexarray(&self, options: &ToCellsOptions) -> Result<CellIndexArray, Error> {
                 self.iter_geo()
                     .map(|v| v.map(Geometry::from))
                     .to_cellindexarray(options)
             }
         }
+    };
+    ($array_type:ty) => {
+        impl<O: OffsetSizeTrait> ToCellListArray<O> for $array_type {
+            fn to_celllistarray(
+                &self,
+                options: &ToCellsOptions,
+            ) -> Result<H3ListArray<CellIndex, O>, Error> {
+                self.iter_geo()
+                    .map(|v| v.map(Geometry::from))
+                    .to_celllistarray(options)
+            }
+        }
 
-        )*
+        impl ToCellIndexArray for $array_type {
+            fn to_cellindexarray(&self, options: &ToCellsOptions) -> Result<CellIndexArray, Error> {
+                self.iter_geo()
+                    .map(|v| v.map(Geometry::from))
+                    .to_cellindexarray(options)
+            }
+        }
     };
 }
 
-impl_from_geoarrow!(
-    geoarrow::array::LineStringArray,
-    geoarrow::array::MultiLineStringArray,
-    geoarrow::array::MultiPointArray,
-    geoarrow::array::MultiPolygonArray,
-    geoarrow::array::PointArray,
-    geoarrow::array::PolygonArray
-);
+impl_to_cells!(geoarrow::array::LineStringArray<O>, O);
+impl_to_cells!(geoarrow::array::MultiLineStringArray<O>, O);
+impl_to_cells!(geoarrow::array::MultiPointArray<O>, O);
+impl_to_cells!(geoarrow::array::MultiPolygonArray<O>, O);
+impl_to_cells!(geoarrow::array::PointArray);
+impl_to_cells!(geoarrow::array::PolygonArray<O>, O);
 
-impl ToCellListArray for WKBArray {
-    fn to_celllistarray(&self, options: &ToCellsOptions) -> Result<H3ListArray<CellIndex>, Error> {
+impl<O: OffsetSizeTrait> ToCellListArray<O> for WKBArray<O> {
+    fn to_celllistarray(
+        &self,
+        options: &ToCellsOptions,
+    ) -> Result<H3ListArray<CellIndex, O>, Error> {
         #[cfg(not(feature = "rayon"))]
-        let pos_iter = (0..self.len()).into_iter();
+        let pos_iter = 0..self.len();
 
         #[cfg(feature = "rayon")]
         let pos_iter = (0..self.len()).into_par_iter();
 
         let cell_vecs = pos_iter
             .map(|pos| {
-                self.get(pos)
-                    .map(|wkb| {
-                        // Geoarrow currently internally only unwraps the parsed geometry
-                        match geozero::wkb::Wkb(wkb.arr.value(wkb.geom_index).to_vec()).to_geo() {
-                            Ok(geom) => geometry_to_cells(&geom, options),
-                            Err(_) => Err(Error::InvalidWKB),
-                        }
-                    })
+                self.get_as_geo(pos)
+                    .map(|geom| geometry_to_cells(&geom, options))
```

Comment:
It would be awesome in the future — if we're able to standardize geometry traits — if h3o is able to implement conversions via those traits. Then you wouldn't have to go through geo objects for all these conversions.

Reply:
There is a lot of potential in the geometry traits. In the case of H3 cells it may be a bit complicated as these can be interpreted as polygons and points.